File: StudentDataLake/StudentDataLake.Common/Entity/Assessments/CheckList.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace StudentDataLake.Common.Entity.Assessments
{
    public enum Level
    {
        ZeroLevel = 0,
    }

    public class CheckList
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string Name { get; set; }

        [Required]
        public Level Level { get; set; }

        [JsonIgnore]
        public virtual List<CheckPoint> CheckPoints { get; set; }
    }
}

File: StudentDataLake/StudentDataLake/StudentDataLakeClient/src/app/models/check-list.ts
import { CheckPoint } from "./check-point";

export enum Level{

    ZeroLevel = 0,

    FirstLevel = 1,

    SecondLevel = 2, 

    ThirdLevel = 3,

    LabLevel = 4,

    JuniorLevel = 5
}

export interface CheckList{

    id: number;

    name: string;

    level: Level;

    checkPoints: CheckPoint[];
}

